Automattic/wp-calypso
# Signup: Fix branching steps if the user comes back from PayPal

#### Changes proposed in this Pull Request

* https://github.com/Automattic/wp-calypso/pull/60181 persists the `excludeSteps` for branching steps, but memoExcludeSteps doesn't. Thus, if the user comes back from 3rd party payment page such as PayPal, back to certain step (intent or starting point) and changes their decision, the flow would be weird.
* So, I refine the usage of the `useBranchSteps` hook to accept `getExcludeSteps` function and it receives the value to decide what steps need to be excluded. Then, we can reset those exclude steps when the user comes back from 3rd party payment

https://user-images.githubusercontent.com/13596067/151792146-6ded35f9-d623-412d-b683-06d9172ecb6a.mp4

#### Testing instructions



* Go to `/start/setup-site?siteSlug=<your_site>` with free plan
* Select `build` intent
* Select `Upgrade Plan` for premium themes
* Select PayPal method and go
* Click cancel to return the checkout modal
* Close the checkout modal and back to intent screen
* Select `write` intent now.
* What you will see
  * Before this PR, you will see the Design Picker step as other steps are excluded and not reset.
  * After this PR, you will see the "give your blog a name" step



Related to https://github.com/Automattic/wp-calypso/pull/60181
